front/components/data_source/DatasourceTablesTabView.tsx

```diff
@@ -128,28 +135,26 @@ export function DatasourceTablesTabView({
         </DataTable.Cell>
       ),
     },
-    {
-      id: "actions",
-      cell: (info: Info) => (
-        <EditOrDeleteDropdown
-          onEdit={() => {
-            setTableToLoad(info.row.original.tableId);
-            setShowTableUploadModal(true);
-          }}
-          onDelete={() => {
-            setTableToLoad(info.row.original.tableId);
-            setShowTableDeleteDialog(true);
-          }}
-        />
-      ),
-    },
   ];
 
   const rows = tables.map((t) => ({
     tableId: t.table_id,
     name: t.name,
     dataSourceId: t.data_source_id,
     timestamp: t.timestamp,
+    moreMenuItems: [
+      {
+        label: "Edit",
+        icon: PencilSquareIcon as React.ComponentType,
```

Comment:
We should not need to cast here 🤔

Reply:
struggled a lot with that .. actually fixing by explicitly typing rows to RowData[]